Support for childOf HierarchyInclusionType in MRCM rules

cmark opened the pull request:
This PR adds a new hierarchy inclusion type, CHILD (corresponding ECL expression is `childOf`), and all the necessary changes to handle it when using `com.b2international.snowowl.snomed.datastore.request.SnomedRequests.prepareGetApplicablePredicates(...)`.

JIRA ticket: https://snowowl.atlassian.net/browse/SO-3346

cmark committed:
[mrcm] add support for childOf HierarchyInclusionType in MRCM rules

cmark committed:
[mrcm] rename CHILDREN enum literal to CHILD

cmark committed:
SO-3346: support CHILD hierarchy inclusion type in SnomedHierarchyDef.

cmark committed:
SO-3346: add childIds field to SnomedConstraintDocument

cmark committed:
SO-3346: use Child option key to properly filter rules by childIds

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRequests.java:
```diff
@@ -468,6 +468,10 @@ public static SnomedConceptSearchRequestBuilder prepareGetSynonyms() {
 					constraintBulkRequestBuilder.add(SnomedRequests.prepareSearchConstraint().all().filterBySelfIds(selfIds));
 				}
 				
+				if (!CompareUtils.isEmpty(ruleParentIds)) {
```
⚠️ Please check the callers of this method: `OpenSnomedConceptInEditor` takes all direct and indirect parents, both stated and inferred, and passes it via this parameter.

cmark replied:
Done!

apeteri commented on snomed/com.b2international.snowowl.snomed.mrcm.model/model/mrcm.ecore:
```diff
@@ -215,6 +215,7 @@
         <details key="documentation" value="Include the specified concept and all its subtype concepts."/>
       </eAnnotations>
     </eLiterals>
+    <eLiterals name="CHILD" value="3" literal="CHILD"/>
```
Please add model documentation to keep it similar to the other literals.

cmark replied:
Done!

cmark committed:
SO-3346: add GET_PARENTS helper function to SnomedConcept

Fix SnomedRequests.prepareGetApplicablePredicates caller in dsv export
test case.

cmark committed:
SO-3346: add documentation to HierarchyInclusionType
